docs/guide/http_cache.md

```diff
@@ -730,11 +737,167 @@ set req.http.X-User-Hash = "b1731d46b0e7a375a5b024e950fdb8d49dd25af85a5c7dd5116a
 
 **5.** Restart the Varnish server and everything should work fine.
 
+##### New anonymous X-User-Hash based on `anonymous_user_id` setting
+
+Since you can configure anonymous user per siteaccess, you need to change the default `.vcl` template to be able to benefit from this functionality. In this example, we assumed that your siteaccesses are matched using `URLElement` matcher. Of course, it works also with other siteaccess matchers. As Varnish is not siteaccess-aware itself, it is recommended to set also different session cookie names per siteaccess.
+
+!!! tip "Different anonymous user per siteaccess"
+    You can set different anonymous user per siteaccess. You can find more information about this setting here: [anonymous_user_id](best_practices.md#anonymous_user_id) 
+
+
+``` yaml
+# ezplatform.yml
+    ezpublish:
+        siteaccess:
+            list: [site, eng, nor]
+            match:
+                URIElement: 1
+        ...
+        system:
+            eng:
+                anonymous_user_id: 15
+                session:
+                    name: eZSESSID_eng
+            nor:
+                anonymous_user_id: 16
+                session:
+                    name: eZSESSID_nor
+        
+```
+
+You need to get the new X-User-Hash for every new anonymous user / anonymous users group. It is done in exactly same way as described above ([User hash generation](http_cache.md#user-hash-generation)), but you have to remember to use siteaccess URI element and additional header which tells eZ which URI should take into account. For instance for `eng` siteaccess:
```

Comment:
`It is done in exactly the same way`
`remember to use the siteaccess URI element and an additional header`
`which URI it should take into account`